web/src/main.js

```diff
@@ -794,6 +794,15 @@ if (!getState().application.isMobile) {
   });
 }
 
+// Production/Consumption
+function toggleElectricityMixMode() {
+  dispatchApplication('electricityMixMode', getState().application.electricityMixMode === 'consumption'
+    ? 'production'
+    : 'consumption');
+  const activeToggle = d3.select('.production-toggle-active-overlay');
```

Comment:
This should be in the observer instead of here.

Reply:
do you mean the activeToggle part?

Reply:
yes

Reply:
done